### PR TITLE
Update to Blazor 3.0 preview

### PR DESCRIPTION
**Pull request by Tarmil**
This is very much WIP not ready to merge.

* Blazor.Extensions.SignalR is not yet ready for 3.0, so template hot reloading is temporarily `#if`'d out.
* Some tests fail.

### CODE REVIEW

**Comment by Tarmil** [comment]
3.0 support in Blazor.Extensions.SignalR was [just merged](https://github.com/BlazorExtensions/SignalR/pull/26) so I should be able to re-enable hot reload on this branch soon.

**Comment by ctaggart** [comment]
Exciting to see this merged! 🎉

File: src/Bolero.Server/Remoting.fs [hunk 1]
```diff
@@ -114,28 +116,27 @@ type internal RemotingService(basePath: PathString, ty: System.Type, handler: ob
...
-                    .Unwrap()
-                    .ContinueWith(fun (tAuthResult: Task<AuthorizationResult>) ->
-                        if tAuthResult.Result.Succeeded then
-                            run()
-                        else
-                            fail ctx
-                            Task.CompletedTask
-                    )
-                    .Unwrap()
+                task {
```
Comment (ctaggart) [inline src/Bolero.Server/Remoting.fs]: I see the switch from `async {}` to `task {}` in this pull request. Is that to line up with the upcoming F# task feature at https://github.com/dotnet/fsharp/pull/6634 ?

Comment (Tarmil) [reply src/Bolero.Server/Remoting.fs]: Yeah, and also simply because it makes the code much nicer than a series of `.ContinueWith(...).Unwrap()`.

**Pull request merged**